### ISSUE TITLE
ISR: Cache invalidation with RegExp

### ISSUE THREAD

**Issue by arsenasatryan** [rx-angular/rx-angular#1777]
# Problem Solved By The Feature

In some cases, on-demand cache invalidation for multiple pages is required simultaneously. It can be useful when using reusable data across multiple or multilanguage pages. As an example, I am using Strapi which can create reusable content types. It allows me to add relations to multiple pages and reuse data across the system. For example,/services-areas/${page1}.../services-areas/${pageN}. the feature will allow you to regenerate those pages simultaneously when the reusable content type changes while keeping other cached pages untouched. 

# Solution

As a solution, I want to suggest adding a new body parameter `(regexpToInvalidate)` to the existing invalidate function or adding a new function for specific regexp invalidation (I am not sure which one will best suit the library architecture). This will allow to search in the cache-store keys via regexp and invalidate them in the background.

**Comment by maxisam** [comment]
fyi, if you use query string for those pages like ?page=1 , you can invalidate them together by setting `allowedQueryParams`. But I can see it is useful to have something like regexpToInvalidate

**Comment by arsenasatryan** [comment]
@maxisam yeah, I know about `allowedQueryParams`. thank you for implementing it, I am already using it (to be honest I was the one who created a feature request for it.). This one is more about invalidation of similar page URLs which somehow depend on something generic. like page/1 page/2, etc.

**Comment by maxisam** [comment]
Yes, I understand. I am just afraid the core team might be swamped right now. You probably won't get that feature for a while. In my case, I can handle it by query string...😅

**Comment by maxisam** [comment]
also I'm glad you are using the implementation!

**Comment by arsenasatryan** [comment]
I will try to implement it independently and create a pull request when I have time. Just I would appreciate your advice on the mentioned approaches of solution. Which one is more preferred in your opinion, as you have more understanding of the codebase you probably can advise the better way.

**Comment by maxisam** [comment]
I probably will try regexpToInvalidate in the post body first. It seems like more flexible to me.

**Comment by maxisam** [comment]
FYI, just want to let you aware there are couple open PR about ISR. You can check out my [repo](https://github.com/maxisam/rx-angular/tree/dev) and see how it will be when all PR is merged
